### PR TITLE
[vue] migrate to jest 27

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/vue/package.json [hunk 1]
```diff
@@ -42,14 +42,14 @@
...
-    "jest": "26.6.3",
+    "jest": "27.0.4",
     "jest-junit": "12.1.0",
     "jest-serializer-vue": "2.0.2",
     "jest-sonar-reporter": "2.0.0",
     "jest-vue-preprocessor": "1.7.1",
     "merge-jsons-webpack-plugin": "1.0.21",
     "mini-css-extract-plugin": "1.6.0",
-    "node-notifier": "10.0.0",
+    "node-notifier": "9.0.1",
```
Comment (atomfrede) [inline generators/client/templates/vue/package.json:52]: What is the reason for the downgrade, any compatibility issues?

Comment (mshima) [reply generators/client/templates/vue/package.json:52]: Peer dependency conflict.

Comment (mshima) [reply generators/client/templates/vue/package.json:52]: Next jest releases allows version 10.

https://github.com/facebook/jest/blob/d1882f2e6033186bd310240add41ffe50c2a9259/packages/jest-core/package.json#L52

Comment (atomfrede) [reply generators/client/templates/vue/package.json:52]: Thanks for the explanation! Are we good to merge with this PR or do you need to add something?

**Review by DanielFran — APPROVED** [review]

**Pull request merged**